core/services/wifi/frontend/index.html

```diff
@@ -0,0 +1,317 @@
+              }
+            })
+        },
+        checkNetworkStatus () {
+          axios({
+            method: 'get',
+            url: `${this.API_URL}/status`,
+            timeout: 500,
+          })
+            .then((response) => {
+              if (response.data.wpa_state === 'COMPLETED'){
+                // Merge status and scan data on connected_network
+                const available_connected_network = this.available_networks.filter((network) => {
+                  return network.ssid === response.data.ssid
+                })[0]
+                this.connected_network = {...response.data, ...available_connected_network}
+              } else {
+                this.connected_network = null
+              }
+              this.wpa_state = response.data.wpa_state.toLowerCase()  
+              this.closeErrorAlert()
+            })
+            .catch(() => {
+              this.openErrorAlert('Could not fetch network status.')
+            })
+        },
+        connectToNetwork () {
+          axios({
+            method: 'post',
+            url: `${this.API_URL}/connect`,
+            timeout: 500,
+            data: {
+              ssid: this.selected_network.ssid,
+              password: document.getElementById('passwordInput').value,
+            },
+          })
+            .then(() => {
+              this.wpa_state = 'connecting'
+            })
+            .catch(() => {
+              this.openErrorAlert(`Could not connect to ${this.selected_network.ssid}.`)
+            })
+            .finally(() => {
+              document.getElementById('passwordInput').value = ''
+            })
+        },
+        disconnectFromNetwork () {
+          axios({
+            method: 'get',
+            url: `${this.API_URL}/disconnect`,
+            timeout: 500,
+          })
+            .then(() => {
+              this.wpa_state = 'disconnecting'
+            })
+            .catch(() => {
+              this.openErrorAlert('Could not disconnect from network.')
+            })
+        },
+        closeErrorAlert () {
+          if (this.alert_box !== null) {
+            if (Metro.infobox.isOpen(this.alert_box)) {
+              Metro.infobox.close(this.alert_box)
+              this.alert_box = null
+            }
+          }
+        },
+        openErrorAlert (message) {
+          this.closeErrorAlert()
+          this.alert_box = Metro.infobox.create(`
+            <div id='alert_box'>
+              <h3>Error</h3>
+              <p>${message}</p>
+            </div>
+            `, 'alert')
+        },
+        openConnectionDialog (network) {
+          this.selected_network = network
+          Metro.dialog.open('#connection_dialog')
+        },
+        openDisconnectionDialog () {
+          Metro.dialog.open('#disconnection_dialog')
+        },
+        signalStrengthClass (network) {
+          if  (network.signallevel >= -60) return 'mif-wifi-full'
```

Comment:
remember that this numbers are in log scale

Reply:
[Found this one](https://www.metageek.com/training/resources/wifi-signal-strength-basics.html). What do you think?

Reply:
```javascript
if (network.signallevel >= -67) return 'mif-wifi-full'
if (network.signallevel >= -70) return 'mif-wifi-mid'
if (network.signallevel >= -80) return 'mif-wifi-low'
return ''
```

Reply:
Looks great, if possible, please add the following table as a commend:
```
| Signal Strength | TL;DR     |  Description                                                                                                                               |
|-----------------|-----------|--------------------------------------------------------------------------------------------------------------------------------------------|
| -30 dBm         | Amazing   | Max achievable signal strength. The client can only be a few feet from the AP to achieve this. Not typical or desirable in the real world. |
| -67 dBm         | Very Good | Minimum signal strength for applications that require very reliable, timely delivery of data packets.                                      |
| -70 dBm         | Okay      | Minimum signal strength for reliable packet delivery.                                                                                      |
| -80 dBm         | Not Good  | Minimum signal strength for basic connectivity. Packet delivery may be unreliable.                                                         |
| -90 dBm         | Unusable  | Approaching or drowning in the noise floor. Any functionality is highly unlikely.                                                          |
Reference: metageek.com/training/resources/wifi-signal-strength-basics.html
```